### PR TITLE
docs : kube-runtime controller params disambiguation

### PR DESCRIPTION
**Pull request by clux**
got a question about possible `ListParams` interactions and realised this could be clarified a little bit.

### CODE REVIEW

File: kube-runtime/src/controller/mod.rs [hunk 1]
```diff
@@ -325,11 +328,17 @@ where
 {
     /// Create a Controller on a type `K`
     ///
-    /// Configure `ListParams` and `Api` so you only get reconcile events
-    /// for the correct `Api` scope (cluster/all/namespaced), or `ListParams` subset
+    /// Takes an [`Api`] object that determines how the `Controller` listens for changes to the `K`.
+    ///
+    /// The [`ListParams`] lets you define a possible subset of objects of `K` that you want to control
+    /// and receive reconcile events for.
+    /// For the full set of objects `K`, you can use `ListParams::default`.
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:335]: Actually this is probably a better wording..

File: kube-runtime/src/controller/mod.rs [hunk 2]
```diff
@@ -325,11 +328,17 @@ where
...
+    ///
+    /// This variant constructor is for [`dynamic`] types found through discovery. Prefer [`Controller::new`] for static types.
     ///
-    /// Unlike `new`, this function accepts `K::DynamicType` so it can be used with dynamic
-    /// resources.
+    /// [`ListParams`]: https://docs.rs/kube/*/kube/api/struct.ListParams.html
+    /// [`Api`]: https://docs.rs/kube/*/kube/api/struct.Api.html
+    /// [`dynamic`]: https://docs.rs/kube/*/kube/core/dynamic/index.html
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:341]: IIRC you can put the type path here instead (such as `kube::core::dynamic`)

File: kube-runtime/src/controller/mod.rs [hunk 3]
```diff
@@ -352,14 +361,16 @@ where
...
-    /// You can customize the parameters used by the underlying `watcher` if
-    /// only a subset of `Child` entries are required.
-    /// The `api` must have the correct scope (cluster/all namespaces, or namespaced)
+    /// Takes an [`Api`] object that determines how the `Controller` listens for changes to the `Child`.
+    /// All owned `Child` objects **must** contain an [`OwnerReference`] pointing back to a `K`.
     ///
-    /// [`OwnerReference`]: https://docs.rs/k8s-openapi/0.10.0/k8s_openapi/apimachinery/pkg/apis/meta/v1/struct.OwnerReference.html
+    /// The [`ListParams`] refer to the possible subset of `Child` objects that you want to own and
+    /// and receive reconcile events for.
+    /// To own the full set of `Child` objects, you can use `ListParams::default`.
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:371]: Ownership is controlled by `OwnerReference`s, this is for filtering out a subset of those owned resources.

File: kube-runtime/src/controller/mod.rs [hunk 4]
```diff
@@ -352,14 +361,16 @@ where
...
+    ///
+    /// [`OwnerReference`]: https://docs.rs/k8s-openapi/0.12.0/k8s_openapi/apimachinery/pkg/apis/meta/v1/struct.OwnerReference.html
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:373]: IIRC you can put the type path here instead

File: kube-runtime/src/controller/mod.rs [hunk 5]
```diff
@@ -373,17 +384,24 @@ where
         self
     }
 
-    /// Indicate an object to watch with a custom mapper
+    /// Specify `Watched` object which `K` has a custom relation to and should be watched
+    ///
+    /// To define the `Watched` relation with `K`, you **must** define a custom relation mapper, which,
+    /// when given a `Watched` object, returns an option or iterator of relevant `ObjectRef<K>` to reconcile.
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:390]: Maybe we can relate this back to `owns` somehow?

File: kube-runtime/src/controller/mod.rs (hunk 2 above)
Comment (clux) [reply kube-runtime/src/controller/mod.rs]: ah, cross-crate links didn't use to work before, but it looks like they do now, changing!

File: kube-runtime/src/controller/mod.rs (hunk 5 above)
Comment (clux) [reply kube-runtime/src/controller/mod.rs:390]: Adding: `If the relation K has to Watched is that K owns Watched, consider using Controller::owns.`

File: kube-runtime/src/controller/mod.rs (hunk 3 above)
Comment (clux) [reply kube-runtime/src/controller/mod.rs:371]: swapping own uses here for track and watch

File: kube-runtime/src/controller/mod.rs [hunk 6]
```diff
@@ -310,8 +310,11 @@ where
 {
     /// Create a Controller on a type `K`
     ///
-    /// Configure `ListParams` and `Api` so you only get reconcile events
-    /// for the correct `Api` scope (cluster/all/namespaced), or `ListParams` subset
+    /// Takes an [`Api`] object that determines how the `Controller` listens for changes to the `K`.
+    ///
+    /// The [`ListParams`] refer to the possible subset of objects of `K` that you want to control
+    /// and receive reconcile events for.
+    /// For the full set of objects `K`, you can use `ListParams::default`.
```
[the remark this answers is not in the document]
Comment (clux) [reply kube-runtime/src/controller/mod.rs:317]: Have clarified the Api/ListParams interaction w.r.g. scope now.

**Review by teozkr — APPROVED** [review]

**Pull request merged**